### PR TITLE
added modularized version of the start of upstream user guide with mi…

### PR DESCRIPTION
**Pull request by mramendi**
…nor changes

### CODE REVIEW

File: doc-content/enterprise-only/optimizer/optimizer-running-the-examples-in-an-IDE-proc.adoc [hunk 1]
```diff
@@ -1,6 +1,9 @@
 [id='optimizer-running-the-examples-in-an-IDE-proc']
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/optimizer-running-the-examples-in-an-IDE-proc.adoc]: Needs cleaning up. I'd do this:
```
= Running the {PLANNER} examples in an IDE (IntelliJ, Eclipse, or Netbeans)
If you use an integrated development environment (IDE), such as IntelliJ, Eclipse, or Netbeans, you can run your downloaded {PLANNER} examples within your development environment.

.Prerequisite
You have downloaded and extracted the {PLANNER} examples from the {PRODUCT} `add-ons.zip` file. For instructions, see the download steps in <<ID for the consolidated "Downloading and running" proc that I suggested.>>.

.Procedure
. Open the file {PLANNER} examples as a new project.
+
--
For IntelliJ or Netbeans, open `examples/sources/pom.xml` as the new project. The Maven integration guides you through the rest of the installation, so you can skip the steps the follow in this procedure.

For Eclipse, open `examples/sources` as the new project.
--
. Add all the JARs to the classpath from the directory `binaries` and the directory `examples/binaries` , except for the file `examples/binaries/optaplanner-examples-*.jar`.
. Add the Java source directory `src/main/java` and the Java resources directory `src/main/resources`.
. Create a run configuration:
+
* Main class: `org.optaplanner.examples.app.OptaPlannerExamplesApp`
* VM parameters (optional): `-Xmx512M -server`
* Working directory: `examples/sources`
. Run the run configuration.
```

File: doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[id='optimizer-running-webexamples-proc']
+= Running the web examples
+Besides the GUI examples, {PRODUCT} also includes a set of web examples for {PLANNER}. The web examples include:
+
+* Vehicle routing: calculating the shortest possible route to pick up all items required for a number of different customers using either http://leafletjs.com/[Leaflet] or http://google.com/maps[Google Maps] visualizations.
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc]: "routing: Calculating ..."

File: doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
+[id='optimizer-running-webexamples-proc']
+= Running the web examples
+Besides the GUI examples, {PRODUCT} also includes a set of web examples for {PLANNER}. The web examples include:
+
+* Vehicle routing: calculating the shortest possible route to pick up all items required for a number of different customers using either http://leafletjs.com/[Leaflet] or http://google.com/maps[Google Maps] visualizations. 
+* Cloud balancing: Assigning processes across computers with different specifications and costs. 
+
+.Prerequisites
+
+You have downloaded and unpacked the {PRODUCT} add-ons package, which includes the examples. For instructions about these actions, see <<examples-download-proc>>.
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc]: You have downloaded and extracted the {PLANNER} examples from the {PRODUCT} `add-ons.zip` file. For instructions, see the download steps in <<ID for the consolidated "Downloading and running" proc that I suggested.>>.

File: doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+The web examples require several JEE APIs to run, such as:
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc]: to run:

File: doc-content/enterprise-only/optimizer/examples-table-ref.adoc [hunk 5]
```diff
@@ -0,0 +1,227 @@
+[id='examples-table-ref']
+= Table of examples
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/examples-table-ref.adoc]: A few tweaks below, plus IBM/RH style guides don't like the type of bulleted list there, prefer something like I've shown:

```
= Table of {PLANNER} examples

Some of the {PLANNER} examples ... for the purpose **[note typo]** of a contest.

A realistic contest is an official, independent contest with the following standards:

* Clearly defined real-world use cases
* Real-world constraints
* Reproducible results ...
* Serious participation from ...

Realistic contests provide ...

.Examples overview [sentence style cap]
[Consider re-ordering either the table or the sections that follow so that they're in a matching order. Odd that they don't correlate.]
```

File: doc-content/enterprise-only/optimizer/ex-machineReassignment-ref.adoc [hunk 6]
```diff
@@ -0,0 +1,59 @@
+[id='ex-machineReassignment-ref']
+= Machine Reassignment (Google ROADEF 2012)
+
+Assign each process to a machine.
+All processes already have an original (unoptimized) assignment.
+Each process requires an amount of each resource (such as CPU, RAM, ...). This is a more complex version of the Cloud Balancing example.
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/ex-machineReassignment-ref.adoc]: (such as CPU and RAM)

File: doc-content/enterprise-only/optimizer/ex-machineReassignment-ref.adoc [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+* Process move cost: A process has a move cost.
+* Service move cost: A service has a move cost.
+* Machine move cost: Moving a process from machine A to machine B has another A-B specific move cost.
+
+The problem is defined by http://challenge.roadef.org/2012/en/[the Google ROADEF/EURO Challenge 2012].
+
+image::UseCasesAndExamples/MachineReassignment/cloudOptimizationIsLikeTetris.png[align="center"]
+
+[[machineReassignmentValueProposition]]
+== Value Proposition
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/ex-machineReassignment-ref.adoc]: .Value proposition (".Label" not "== Heading", and sentence style not headline)

File: doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+64queens  has  64 queens with a search space of 10^115.
+256queens has 256 queens with a search space of 10^616.
+----
+
+The implementation of the _n_ queens example has not been optimized because it functions as a beginner example. Nevertheless, it can easily handle 64 queens.
+With a few changes it has been shown to easily handle 5000 queens and more.
+
+
+[[nQueensDomainModel]]
+.Domain Model
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc]: Sub-heading, not label. Could be it's own module (hence why it's anchored), but for the sake of time, sub-heading within the module is fine for now:
```
== N Queens domain model

The N Queens example uses...
```

File: doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+This example uses the domain model to solve the four queens problem.
+
+. *Creating a Domain Model*
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc]: Change these sections to bulleted instead of numbers, and make sure the line breaks are consistent (first doesn't have line break in output, others do).

File: doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+----
++
+. *Finding the Solution*
++
+A single `NQueens` instance contains a list of all `Queen` instances.
+It is the `Solution` implementation which will be supplied to, solved by, and retrieved from the Solver.
+
+Notice that in the four queens example, NQueens's `getN()` method will always return four.
+
+.A Solution for Four Queens Shown in the Domain Model
```
Comment (sterobin) [inline doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc]: The table is nearly unusable in the output because there are too many columns and the last two columns have long, crazily wrapped heads. I'd remove the "A solution" column and content and make it stand alone, followed by the now 5-columned table that's at least readable now.
```
.A solution for four ...
[image]

.Solution details
[cols="6a,1,1,1,1", options="header"]
 |===
 | Queen
 | columnIndex
 | rowIndex
 | ascendingDiagonalIndex (columnIndex + rowIndex)
 | descendingDiagonalIndex (columnIndex - rowIndex)

etc.
.....
```

File: assemblies/assembly_business-optimizer-gsg/main.adoc [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+:imagesdir: _images/
+:context: business-optimizer
+include::_artifacts/author-group.adoc[]
+
+// Purpose statement
+As a business rules developer, you can use the {PLANNER} to find the optimal solution to planning problems based on a set of limited resources and under specific constraints.
+
+Use this document to start developing applications with {PLANNER}.
+
+include::{enterprise-dir}/optimizer/optimizer-about-business-optimizer-con.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (mramendi) [reply assemblies/assembly_business-optimizer-gsg/main.adoc]: I do not want to use the term "NP-complete_ness_" without making sure this is how people actually use it.

File: _artifacts/document-attributes-dm.adoc [hunk 12]
```diff
@@ -27,3 +27,4 @@
```
[the remark this answers is not in the document]
Comment (mramendi) [reply _artifacts/document-attributes-dm.adoc]: We definitely have no resources to change any diagrams now. Nor to get any new labels reviewed.

File: doc-content/enterprise-only/optimizer/cloudbal-helloworld-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+[source,java,options="nowrap"]
+----
+        SolverFactory<CloudBalance> solverFactory = SolverFactory.createFromXmlResource(
+                "org/optaplanner/examples/cloudbalancing/solver/cloudBalancingSolverConfig.xml");
+        Solver solver<CloudBalance> = solverFactory.buildSolver();
+----
+
+. Load the problem. 
++
+`CloudBalancingGenerator` generates a random problem: you will replace this with a class that loads a real problem, for example from a database.
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/cloudbal-helloworld-proc.adoc]: "a real problem, such as from a database" does not sound right to me and Geoffrey;s text does sound right to me here

File: doc-content/enterprise-only/optimizer/cloudbal-solver-configuraton-ref.adoc [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+  <termination>
+    <secondsSpentLimit>30</secondsSpentLimit>
+  </termination>
+</solver>
+----
+====
+
+This solver configuration consists of three parts:
+
+. *Domain model configuration*: _What can Planner change?_
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/cloudbal-solver-configuraton-ref.adoc]: As far as I understand, this is also about the order of the parts in the file, so numbers do make sense here. Absent SME input I would not change. And we are not getting SME input now.

File: doc-content/enterprise-only/optimizer/cloudbal-score-drools-proc.adoc [hunk 15]
```diff
@@ -0,0 +1,79 @@
+[id='cloudbal-score-drools-proc']
+= Configuring score calculation using Drools
+
+You can use the Drools language to define constraints. Drools score calculation uses incremental calculation, where every score constraint is written as one or more score rules. 
+
+Using the Drools rule engine for score calculation enables you to integrate with other Drools technologies, such as decision tables (XLS or web based), the KIE Workbench, and so on.
+
+.Procedure
+To use the Drools rule engine as a score function, simply add a `scoreDrl` resource in the classpath. In the `cloudBalancingSolverConfig.xml` file, add or uncomment the setting:
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/cloudbal-score-drools-proc.adoc]: Cleaning up the steps to this extent defeats the purpose of the tutorial, where the explanations is as important as the steps themselves. Making a bit of change but definitely not that extensive.

File: doc-content/enterprise-only/optimizer/optimizer-running-the-examples-proc.adoc [hunk 16]
```diff
@@ -1,14 +1,12 @@
 [id='optimizer-running-the-examples-proc']
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/optimizer-running-the-examples-proc.adoc]: No, because "downloading" is referenced in several places. That is the reason I broke it out.

File: doc-content/enterprise-only/optimizer/examples-download-run-con.adoc [hunk 17]
```diff
@@ -0,0 +1,3 @@
+[id='examples-download-run-con']
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/examples-download-run-con.adoc]: Not all downloaded examples are run like this; there are also webexamples. And downloading applies to all of them. Also, one might opt to run using an IDE, but downloading still applies. This is why downloading is a separate procedure.

File: doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc (hunk 3 above)
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc]: Consolidating two procedures into one then referencing *some of the steps* does not sound like a good decision to me.

File: doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc (hunk 4 above)
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc]: Removing "such as" would mean the list is complete. Not doing without SME input.

File: doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc (hunk 8 above)
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc]: I am definitely lost in why you think some things should be sub-headings instead of labels. I made everything labels, well that was the intention, one or two slipped through.

File: _artifacts/document-attributes-dm.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply _artifacts/document-attributes-dm.adoc]: As I said, "this is obviously not a task for this release".

File: assemblies/assembly_business-optimizer-gsg/main.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply assemblies/assembly_business-optimizer-gsg/main.adoc]: That's how the term is used in the external wiki resource that that section links to, just fyi.

File: doc-content/enterprise-only/optimizer/cloudbal-helloworld-proc.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply doc-content/enterprise-only/optimizer/cloudbal-helloworld-proc.adoc]: Sure

File: doc-content/enterprise-only/optimizer/cloudbal-score-drools-proc.adoc (hunk 15 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply doc-content/enterprise-only/optimizer/cloudbal-score-drools-proc.adoc]: I'm not saying to remove the explanations. I'm saying that the steps need to be clear action-based steps, so I was proposing to cut through what I viewed as fluff. Also, the bigger issue was the the steps had formatting issues with indentation and incorrect numbering sequence, which I assume you fixed but which is still incorrect in the output in the JIRA. I'll assume that output isn't the latest.

File: doc-content/enterprise-only/optimizer/examples-download-run-con.adoc (hunk 17 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply doc-content/enterprise-only/optimizer/examples-download-run-con.adoc]: Yes, I know why you did it. It just made it overly choppy imo and it made sense for there to be at least a steamlined process of downloading and running that would apply to most, and then they could use another running procedure as applicable. But this is just subjective opinion so fine what you have.

File: doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc (hunk 3 above)
Comment (sterobin) [reply doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc]: Agreed. Overly breaking down procedures didn't seem great to me either. So just gains and losses of both.

File: doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc (hunk 4 above)
Comment (sterobin) [reply doc-content/enterprise-only/optimizer/optimizer-running-webexamples-proc.adoc]: Good point.

File: doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc (hunk 8 above)
Comment (sterobin) [reply doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc]: Per mode doc convention, headings are for standalone, typically non-repeated content, such as concepts, tasks, and references. Usually they're their own modules, but concepts and refs can technically contain sub-headings too, but I've found that almost always a sub-headed section can be its own module as well that's just included. Labels are used for elements like "Procedure," "Prerequisites," "Additional resources," "Table," "Figure" (the last two automatically placed by asciidoc), or code snippet labels. Labeled content is typically repeated types of content within headed sections, but not always (like code snippet labels).

In short, you use labels to label a block of content, not to head a section. This example is clearly a section of content, not a block to be labeled, and thus makes more sense as a sub-heading.

File: doc-content/enterprise-only/optimizer/cloudbal-solver-configuraton-ref.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply doc-content/enterprise-only/optimizer/cloudbal-solver-configuraton-ref.adoc]: Understood.

File: assemblies/assembly_business-optimizer-gsg/main.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (mramendi) [reply assemblies/assembly_business-optimizer-gsg/main.adoc]: Ah - makes sense, I can play with it then

File: doc-content/enterprise-only/optimizer/cloudbal-score-drools-proc.adoc (hunk 15 above)
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/cloudbal-score-drools-proc.adoc]: I now see there were lingering issues, did something :) reuploaded the built doc; PR to be  updated once I go thru all comments

File: doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc (hunk 8 above)
Comment (mramendi) [reply doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc]: I have attempted to implement what you write here, which resulted in "Domain model" being a heading in some cases and a section in others. When I push the changes (won't reupload right now) please review; we might have to leave it be right now?

Comment (sterobin) [reply doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc]: You can leave anything be right now that you wish. We are admittedly at the end of the road, and that's okay. That's the purpose of Agile, to get things out and iterate to improve. Can't do everything perfectly every time always.
